dojo/tools/intsights/parser.py

```diff
@@ -0,0 +1,105 @@
+import json
+from urllib.parse import urlparse
+
+from dojo.models import Endpoint, Finding
+
+
+class IntSightsParser(object):
+    """
+    IntSights Threat Intelligence Feed
+    """
+
+    def get_scan_types(self):
+        return ["IntSights Scan"]
+
+    def get_label_for_scan_types(self, scan_type):
+        return "IntSights Scan"
+
+    def get_description_for_scan_types(self, scan_type):
+        return "IntSights report file can be imported in JSON format."
+
+    def get_findings(self, file, test):
+        duplicates = dict()
+
+        if not file:
+            return []
+
+        data = file.read()
+
+        try:
+            findings = json.loads(str(data, 'utf-8'))
+        except:
+            findings = json.loads(data)
+
+        if not findings:
+            raise ValueError('IntSights report contains errors: No vulnerabilities were found in the data provided')
+
+        if findings.get('Findings'):
+            for finding in findings['Findings']:
+                unique_id_from_tool = finding['_id']
+                title = finding['Details']['Title']
+                description = f'{finding["Details"]["Description"]}' \
+                              f'\r\n\r\n----' \
+                              f'\r\n\r\n**Type**: {finding["Details"]["Type"]}' \
+                              f'\r\n**SubType**: {finding["Details"]["SubType"]}' \
+                              f'\r\n**Source**: {finding["Details"]["Source"]["URL"]}' \
+                              f'\r\n**Source Type**: {finding["Details"]["Source"]["Type"]}' \
+                              f'\r\n**Source Network Type**: {finding["Details"]["Source"]["NetworkType"]}' \
+                              f'\r\n\r\n----' \
+                              f'\r\n**Asset Type**: {finding["Assets"][0]["Type"]}' \
+                              f'\r\n\r\n----' \
+                              f'\r\n**Takedown Status**: {finding["TakedownStatus"]}'
+                severity = finding['Details']['Severity']
+                mitigation = "N/A"
+                impact = "N/A"
+                references = finding["Details"]["Source"]["URL"]
```

Comment:
is this field always present?

Reply:
It is present in all of the findings I have data for, but I will change it to fail safe. Will also rebase off of dev.